examples/features/customloadbalancer/client/customroundrobin/customroundrobin.go

```diff
@@ -0,0 +1,249 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package customroundrobin
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+func init() {
+	balancer.Register(customRoundRobinBuilder{})
+}
+
+const customRRName = "custom_round_robin"
+
+type customRRConfig struct {
+	serviceconfig.LoadBalancingConfig `json:"-"`
+
+	// N represents how often pick iterations chose the second SubConn in the
+	// list. Defaults to 3. If 0 never choses second SubConn.
+	N uint32 `json:"n,omitempty"`
+}
+
+type customRoundRobinBuilder struct{}
+
+func (customRoundRobinBuilder) ParseConfig(s json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+	lbConfig := &customRRConfig{
+		N: 3,
+	}
+	if err := json.Unmarshal(s, lbConfig); err != nil {
+		return nil, fmt.Errorf("custom-round-robin: unable to unmarshal customRRConfig: %v", err)
+	}
+	return lbConfig, nil
+}
+
+func (customRoundRobinBuilder) Name() string {
+	return customRRName
+}
+
+func (customRoundRobinBuilder) Build(cc balancer.ClientConn, bOpts balancer.BuildOptions) balancer.Balancer {
+	pfBuilder := balancer.Get(grpc.PickFirstBalancerName)
+	if pfBuilder == nil {
+		return nil
+	}
+	return &customRoundRobin{
+		cc:               cc,
+		bOpts:            bOpts,
+		pfs:              resolver.NewEndpointMap(),
+		pickFirstBuilder: pfBuilder,
+	}
+}
+
+var logger = grpclog.Component("example")
+
+type customRoundRobin struct {
+	// All state and operations on this balancer are either initialized at build
+	// time and read only after, or are only accessed as part of it's
+	// balancer.Balancer API (UpdateState from children only comes in from
+	// balancer.Balancer calls as well, and children are called one at a time),
+	// in which calls are guaranteed to come synchronously. Thus, no extra
+	// synchronization is required in this balancer.
+	cc    balancer.ClientConn
+	bOpts balancer.BuildOptions
+	// Note that this balancer is a petiole policy which wraps pick first (see
+	// gRFC A61). This is the intended way a user written custom lb should be
+	// specified, as pick first will contain a lot of useful functionality, such
+	// as Sticky Transient Failure, Happy Eyeballs, and Health Checking.
+	pickFirstBuilder balancer.Builder
+	pfs              *resolver.EndpointMap
+
+	n                    uint32
+	inhibitPickerUpdates bool
+}
+
+func (crr *customRoundRobin) UpdateClientConnState(state balancer.ClientConnState) error {
+	if logger.V(2) {
+		logger.Info("custom_round_robin: got new ClientConn state: ", state)
+	}
```

Comment:
Deleted this one. Left the error log from UpdateSubConnState since it signals to the reader that that is deprecated and shouldn't be called. Should I leave that one too?